unittests/windowstests.py

```diff
@@ -398,3 +398,44 @@ def test_vsenv_option(self):
         with mock.patch.object(self, 'install_command', self.meson_command + ['install']):
             out = self.install(override_envvars=env)
             self.assertIn('Activating VS', out)
+
+    @skip_if_not_language('nasm')
+    def test_nasm_linker_args(self):
+        # TODO: there should be some way to query how we're linking things
+        # without resorting to reading the ninja.build file
+        if self.backend is not Backend.ninja:
+            raise SkipTest('This test reads the ninja file')
+
+        testdir = os.path.join(self.nasm_test_dir, '3 nasm only')
+        self.init(testdir)
+
+        env = get_fake_env(testdir, self.builddir, self.prefix)
+        cc = detect_c_compiler(env, MachineChoice.HOST)
+
+        build_ninja = os.path.join(self.builddir, 'build.ninja')
+        with open(build_ninja, encoding='utf-8') as f:
+            contents = f.read()
+
+        self.assertRegex(contents, r'build dummy(\.dll)?.*: nasm_LINKER')
+        if cc.get_id() == 'msvc':
+            # The following test DLL defaultlibs
+            # TODO: add an example for compiling a Windows executable from NASM
+            self.assertRegex(contents, r'LINK_ARGS =(.+)"/DEFAULTLIB:ucrt(d)?\.lib"')
+            self.assertRegex(contents, r'LINK_ARGS =(.+)".*/DEFAULTLIB:vcruntime(d)?\.lib"')
+            self.assertRegex(contents, r'LINK_ARGS =(.+)"/DEFAULTLIB:msvcrt(d)?\.lib"')
```

Comment:
Is this unit test needed? If those libs are missing it will fail to link anyway, right?

Reply:
Yes: https://github.com/mesonbuild/meson/pull/11127#discussion_r1039652129

>We definitely miss a unit test that mix C and NASM files, we need to make sure that when mixing those languages, it uses c_link_args and not nasm_link_args option

The bits in which I check the `defaultlib` lines are defensive programming on my end, let me know and I'll remove them.